test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,6 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
```

Comment:
Does deleting this line improve the test? Will this test confirm that the saving happens _only_ for mutating commands?